python/tk_multi_workfiles/file_model.py

```diff
@@ -983,35 +981,11 @@ def _on_finder_work_area_resolved(self, search_id, work_area):
             # ignore result
             return
 
-        search = self._in_progress_searches[search_id]
-        entity_key = self._gen_entity_key(search.entity)
-
-        # keep track of work area for this entity:
-        self._entity_work_area_map[entity_key] = work_area
-
-        # consolidate list of sandbox users for all current work areas:
-        have_user_sandboxes = False
-        users_by_id = {}
-        handled_entities = set()
-        for search in self._in_progress_searches.values():
```

Comment:
This is the main piece of code that was wrong before. It accumulated the sandboxes for all contexts that were being scanned. I can't understand why it was written this way, but I definitely say the the logic was wrong because the intent is to accumulate the list of all users across sandboxes and this accumulates the list of all users across sandboxes that are being scanned. As jobs complete, this list shortens and less and less sandboxes will be returned. The new code simply broadcasts the users that were found.